mqtt/mqtt-bridge/src/messages.rs

```diff
@@ -81,20 +83,38 @@ impl<S> StoreMqttEventHandler<S> {
 
     fn transform(&self, topic_name: &str) -> Option<String> {
         self.topic_mappers.values().find_map(|mapper| {
-            mapper
-                .topic_settings
-                .in_prefix()
-                // maps if local does not have a value it uses the topic that was received,
-                // else it checks that the received topic starts with local prefix and removes the local prefix
-                .map_or(Some(topic_name), |local_prefix| {
-                    topic_name.strip_prefix(format!("{}/", local_prefix).as_str())
-                })
-                // match topic without local prefix with the topic filter pattern
-                .filter(|stripped_topic| mapper.topic_filter.matches(stripped_topic))
-                .map(|stripped_topic| match mapper.topic_settings.out_prefix() {
-                    Some(remote_prefix) => format!("{}/{}", remote_prefix, stripped_topic),
-                    None => stripped_topic.to_string(),
-                })
+            if mapper.topic_filter.matches(topic_name) {
+                mapper
+                    .topic_settings
+                    .in_prefix()
+                    // maps if local does not have a value it uses the topic that was received,
+                    // else it checks that the received topic starts with local prefix and removes the local prefix
+                    .map_or(Some(topic_name), |local_prefix| {
+                        if mapper.topic_settings.out_prefix().is_none() {
+                            // inPrefix is not empty which means a topic separator was appended <inPrefix>/<topic>
+                            // so it needs to be removed if there is no out prefix
+                            topic_name.strip_prefix::<&str>(
+                                format!("{}{}", local_prefix, TOPIC_SEPARATOR).as_ref(),
+                            )
+                        } else {
+                            topic_name.strip_prefix::<&str>(local_prefix)
+                        }
+                    })
+                    .map(|stripped_topic| match mapper.topic_settings.out_prefix() {
+                        Some(remote_prefix) => {
+                            if mapper.topic_settings.in_prefix().is_none() {
```

Comment:
I am confused - why do you need this logic to strip/add the topic separator? 
Why can't you always strip the '/' and add it back?

Reply:
I changed the code to make it more clear. It can't strip / all the time because if it was not added then strip_prefix will return None, also there is a case when topic is /temp and inPrefix and outPrefix are empty so it should not be stripped.

Reply:
Thanks Anca.. but I am not sure I still follow. 
What happens in this case - 
InPrefix: ""
OutPrefix: "/foo"
Topic: "/telemetry"
Will you end up with an extra '/' in the output topic?

Reply:
yes, it will set topic to /foo//telemetry
same for:
if InPrefix: "/in"
outPrefix: "/foo"
topic: "/telemetry"
it subscribes to /in//telemetry and outputs to /foo//telemetry because it doesn't strip / if it is included in topic

Reply:
But that is not correct, right?
For this - 
if InPrefix: "/in"
outPrefix: "/foo"
topic: "/telemetry"
I expect the sub to be on /in/telemetry and pub on /foo/telemetry. Otherwise we should explicitly say that we will add a '/' in the middle.

Reply:
I removed the default topic separator, now it expects to have:

topic: "bla"
inPrefix: "foo/"
outPrefix: "bar/"

subscribes to "foo/bla" and sends to "bar/bla"